src/modules/convert/GEN_STRUCT_DEMUX_fbt.cpp

```diff
@@ -93,9 +87,15 @@ bool GEN_STRUCT_DEMUX::createInterfaceSpec(const char *paConfigString, SFBInterf
 
       size_t structSize = structInstance->getStructSize();
       if(structSize != 0 && structSize < cgInvalidPortId) { //the structure size must be non zero and less than cgInvalidPortId (maximum number of data outputs)
-        CStringDictionary::TStringId *doDataTypeNames = new CStringDictionary::TStringId[GEN_STRUCT_MUX::calcStructTypeNameSize(*structInstance)];
-        CStringDictionary::TStringId *doNames = new CStringDictionary::TStringId[structSize];
-        CStringDictionary::TStringId *diDataTypeNames = new CStringDictionary::TStringId[1];
+       
+        // we need doDataTypeNames here because we need to call fillDataPointSpec later which requires a non-const reference of the pointer
+        // wich we cannot get from the unique_ptr
+        auto doDataTypeNames = new CStringDictionary::TStringId[GEN_STRUCT_MUX::calcStructTypeNameSize(*structInstance)];
+        mDoDataTypeNames.reset(doDataTypeNames);
```

Comment:
I would suggest doing it the other way around by assigning a local variable from the `unique_ptr` and also moving it directly before the call to `fillDataPointSpec` below, since the pointer should not be used for anything else after `fillDataPointSpec` has modified it.